plugins/modules/consul_agent_check.py

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, Michael Ilg
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: consul_agent_check
+short_description: Add, modify & delete checks within a consul cluster
+description:
+ - Registers checks for an agent with a consul cluster.
+   A service is some process running on the agent node that should be advertised by
+   consul's discovery mechanism. It may optionally supply a check definition,
+   a periodic service test to notify the consul cluster of service's health.
+ - "Checks may also be registered per node e.g. disk usage, or cpu usage and
```

Comment:
```suggestion
 - "Checks may also be registered per node, for example disk usage, or cpu usage and
```
(https://docs.ansible.com/ansible/devel/dev_guide/style_guide/#avoid-using-latin-phrases)